packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -262,13 +271,28 @@ class CupertinoPageRoute<T> extends PageRoute<T> {
     return result;
   }
 
+  static Animation<double> _animationControllerForTransitionAnimation(Animation<double> animation) {
+    Animation<double> controller = animation;
+    while(controller is ProxyAnimation || controller is TrainHoppingAnimation) {
+      final Animation<double> proxy = controller;
```

Comment:
This works because we believe that the implementation only creates these two animation types? If someone changed the implementation and started using FooAnimationWrapper or, worse still _FooAnimationWrapper, we'd just silently fail, right?

Reply:
Took a different approach